libcob/coblocal.h

```diff
@@ -555,19 +554,27 @@ DECLNORET COB_HIDDEN void	cob_hard_failure (void) COB_A_NORETURN;
 
 /* static inline of smaller helpers */
 
+/* The following functions are only defined on-demand using flags
+   COB_NEEDS_*. Otherwise, we get either defined-but-not-used
+   warnings, or not-always-inlined warnings. */
```

Comment:
I haven't seen those warnings, do you have that locally / in CI?
Any "special flags" used? Which C compiler version?

Reply:
Hum... ok, this is getting weird:

Reply:
Hum... ok, this is getting weird:
```
#include "coblocal.h"
#include "config.h"
```
triggers warnings such as:
```
In file included from ../../libcob/profiling.c:29:
../../libcob/coblocal.h:577:1: warning: ‘always_inline’ function might not be inlinable [-Wattributes]
  577 | cob_max_int (const int x, const int y)
      | ^~~~~~~~~~~
../../libcob/coblocal.h:570:1: warning: ‘always_inline’ function might not be inlinable [-Wattributes]
  570 | cob_min_int (const int x, const int y)
      | ^~~~~~~~~~~
../../libcob/coblocal.h:577:1: warning: ‘cob_max_int’ defined but not used [-Wunused-function]
  577 | cob_max_int (const int x, const int y)
      | ^~~~~~~~~~~
../../libcob/coblocal.h:570:1: warning: ‘cob_min_int’ defined but not used [-Wunused-function]
  570 | cob_min_int (const int x, const int y)
      | ^~~~~~~~~~~
```

Whereas:
```
#include "config.h"
#include "coblocal.h"
```
does not.

There must be some flag in "config.h" that is needed. 

Maybe "coblocal.h" should have a `#include "config.h"` instead of these changes ?

Reply:
No. config.h should just always be included first.